modules/pmi/pyext/src/restraints/residue_proximity_restraint.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import print_function
+import IMP
+import IMP.core
+import IMP.algebra
+import IMP.atom
+import IMP.container
+import IMP.pmi.tools
+import IMP.pmi.restraints
+
+import math
+import numpy as np
+
+class ResidueProteinProximityRestraint(IMP.pmi.restraints.RestraintBase):
+
+    """ Restraint residue/residues to bind to unknown location in a target"""
+
+    def __init__(self,
+                 hier,
+                 selection,
+                 cutoff = 6.,
+                 sigma = 3.,
+                 xi = 0.01,
+                 resolution=1.0,
+                 weight = 1.0,
+                 label = None):
+        """
+        Constructor
+        @ param hier        Hierarchy of the system
+        @ param section     Selection of residues and target
+                            syntaxis (prot, r1, r2, target_prot)  or  
+                            (prot1, r1, r2, target_prot, target_r1, target_r2) 
+        @ param cutoff      Distance cutoff between selected segment and target
+                            protein 
+        @ param sigma       Distance variance between selected fragments
+        @ param xi          Slope of a distance-linear scoring function that funnels 
+                            the score when the particles are too far away
+        @ param resolution  Resolution at which apply restraint
+        @ param weight      Weight of the restraint
+        @ param label       Extra text to label the restraint so that it is
+                            searchable in the output 
+        """
+        self.hier = hier
+        m = self.hier.get_model()
+
+        rname = "ResidueProteinProximityRestraint"
+        super(ResidueProteinProximityRestraint, self).__init__(
+            m, name="ResidueProteinProximityRestraint", label=label, weight=weight)
+
+        self.cutoff = cutoff
+        self.sigma = sigma
+        self.xi = xi
+        self.resolution = resolution
+
+        # Check selection
+        print('selection', selection, isinstance(selection, tuple))
+        if (not isinstance(selection, tuple)) and (not isinstance(selection, list)):
+            raise ValueError("Selection should be a tuple or list")
+        if len(selection) < 4:
+            raise ValueError("Selection should be (prot, r1, r2, target_prot) or \
+                             (prot1, r1, r2, target_prot, target_r1, target_r2) ")
```

Comment:
Isn't there a PMI function already to parse a selection tuple, `IMP.pmi.tools.select_by_tuple_2` perhaps? If not, I should write one!

Reply:
Yes, but it select from a single protein (start,stop,molname,copynum,statenum). Here the tuple select from two different proteins.